src/qw/remote_repo/_github.py

```diff
@@ -71,18 +142,115 @@ def __init__(self, conf):
     def _get_token(self):
         return keyring.get_password("qw", f"{self.username}/{self.reponame}")
 
-    def get_issue(self, number: int):
+    def _graph_ql(self, query):
+        """Execute a GraphQL query, returning the response."""
+        url = self.gh.session.build_url("graphql")
+        return self.gh.session.request(
+            "POST",
+            url,
+            json={
+                "query": query,
+            },
+        )
+
+    def get_issue(self, number: int) -> Issue:
         """Get the issue with the specified number."""
         issue = self.gh.issue(self.username, self.reponame, number)
         return Issue(issue)
 
+    def get_pull_request(self, number: int) -> PullRequest | None:
+        """Get the pull request with the specified number."""
+        response = self._graph_ql(
+            f"""query {{
+repository(owner: "{self.username}", name: "{self.reponame}") {{
+    pullRequest(number: {number}) {{
+        body
+        closed
+        closedAt
+        closingIssuesReferences(first: 100) {{
+            nodes {{
+                number
+            }}
+        }}
+        isDraft
+        labels(last: 100) {{
+            nodes {{
+                name
+            }}
+        }}
+        merged
+        number
+        title
+    }}
+}}
+}}""",
+        )
+        if not status_is_ok(response.status_code):
+            logger.info(
+                "Failed ({}) to get the closing numbers for issue {}",
+                response.status_code,
+                number,
+            )
+            return None
+        result = json.loads(response.content)
+        return PullRequest(**result["data"]["repository"]["pullRequest"])
+
     @property
     def issues(self):
         """Get all issues for the repository."""
         return [
-            Issue(issue) for issue in self.gh.issues_on(self.username, self.reponame)
+            Issue(issue)
+            for issue in self.gh.issues_on(
+                self.username,
+                self.reponame,
+            )
```

Comment:
I _think_ this will currently also get all pull requests as well? Or am I wrong there?